lib/rspec/core/formatters/documentation_formatter.rb

```diff
@@ -6,12 +6,17 @@ module Core
     module Formatters
       # @private
       class DocumentationFormatter < BaseTextFormatter
-        Formatters.register self, :example_group_started, :example_group_finished,
+        Formatters.register self, :example_started, :example_group_started, :example_group_finished,
                             :example_passed, :example_pending, :example_failed
 
         def initialize(output)
           super
           @group_level = 0
+          @messages = nil
```

Comment:
Rather than nil, I think I'd rather this was always an array, which is cleared to reduce allocations. Instead we should record wether exampes are running or not e.g:
```Ruby
@messages = []
@examples_running = false
```

Reply:
This this just purely a performance change, or do you prefer the semantics of it? Because I was going for simplicity - if the `@messages` buffer exists, please use it, rather than having a separate toggle which could get out of sync.

Reply:
Both, I prefer it, and its more performant with large test suites

Reply:
What about if we defer the allocation of messages until it's actually needed? i.e. it stays nil until someone actually tries to log a message? Then, it's zero overhead.

Reply:
From a code style pov I don't like that, as it means declaring it as an array on first use, then clearing it later, so its an extra if, 1 allocation isn't an issue, its that currently its 1 allocation per example group